source/server/filter_chain_manager_impl.h

```diff
@@ -13,26 +15,29 @@
 namespace Envoy {
 namespace Server {
 
+class FilterChainFactoryBuilder {
+public:
+  virtual ~FilterChainFactoryBuilder() = default;
+  virtual std::unique_ptr<Network::FilterChain>
+  buildFilterChain(const ::envoy::api::v2::listener::FilterChain& filter_chain) const = 0;
+};
+
 /**
  * Implementation of FilterChainManager.
  */
 class FilterChainManagerImpl : public Network::FilterChainManager,
                                Logger::Loggable<Logger::Id::config> {
 public:
+  explicit FilterChainManagerImpl(Network::Address::InstanceConstSharedPtr address)
+      : address_(address) {}
+
   // Network::FilterChainManager
   const Network::FilterChain*
   findFilterChain(const Network::ConnectionSocket& socket) const override;
+
   void
-  addFilterChain(uint16_t destination_port, const std::vector<std::string>& destination_ips,
-                 const std::vector<std::string>& server_names,
-                 const std::string& transport_protocol,
-                 const std::vector<std::string>& application_protocols,
-                 const envoy::api::v2::listener::FilterChainMatch_ConnectionSourceType source_type,
-                 const std::vector<std::string>& source_ips,
-                 const Protobuf::RepeatedField<Protobuf::uint32>& source_ports,
-                 Network::TransportSocketFactoryPtr&& transport_socket_factory,
-                 std::vector<Network::FilterFactoryCb> filters_factory);
-  void finishFilterChain() { convertIPsToTries(); }
+  addFilterChain(absl::Span<const ::envoy::api::v2::listener::FilterChain* const> filter_chain_span,
```

Comment:
I'm not sure it's really worth converting this to a span? Can we just pass the relevant proto container directly via const reference?

Reply:
The idea is to provide as few information as possible, and my limited knowledge on protobuf structure :)

By `fewer message` my point is using span so we can use "vector of  FilterChain" in test and "a proto container" in prod. They should be functional equivalent from the perspective of this method. So is the performance from the declaration of `absl::Span`

Reply:
OK that's fine.